server/aesdsocket.c

```diff
@@ -0,0 +1,394 @@
+        return -1;
+    }
+#ifdef DEBUG
+    struct sockaddr client_addr;
+    socklen_t addr_size = sizeof(client_addr);
+
+    new_fd = accept(sd , &client_addr, &addr_size);
+    // Error check 
+    if (new_fd == -1){
+        perror("accept");
+        return -1;
+    }
+
+    // Print IP address
+    // The inet_toa function requires a sockaddr_in structure
+    // Reference: https://stackoverflow.com/questions/3060950/how-to-get-ip-address-from-sock-structure-in-c
+    struct sockaddr_in * ip_client = (struct sockaddr_in *)&client_addr;
+    syslog(LOG_INFO , "Accepted connection from %s\n", inet_ntoa(ip_client->sin_addr));
+    
+    // Unit test to see if receive works from the sockettest.sh
+    char buffer[10];
+
+    int no_bytes = recv(new_fd , buffer , sizeof(buffer) , 0);
+
+    printf("Printing the number of received characters: %d\n", no_bytes);
+    printf("Error code: %d\n", errno);
+
+    for(int i = 0; i < no_bytes; i++){
+        printf("%c", buffer[i]);
+    }
+#endif
+    
+    freeaddrinfo(servinfo);
+
+    // Logic to receive data packets
+
+    struct sockaddr client_addr;
+    socklen_t addr_size = sizeof(client_addr);
+
+    // Declare a static buffers: One to get data from recv() 
+    char recv_buffer[BUFFER_SIZE];
+
+    // Malloc buffer of the same size initially for sending the data over to the client and writing to teh file
+    write_buffer = (char *)malloc(sizeof(char) * BUFFER_SIZE);
+    // Error check: If malloc failed
+    if (write_buffer == NULL){
+        syslog(LOG_ERR , "malloc\n");
+        return -1;
+    }
+
+    // Flag to check if \n was received
+    uint8_t newline_flag = 0;
+    int byte_count = 0;
+    int datapacket = 0;
+    int final_size = 1;
+    int realloc_int = 0;
+    static int file_size = 0;
+
+    // While any signal has not been encountered yet
+    while(1){
+
+        new_fd = accept(sd , &client_addr, &addr_size);
+        // Error check 
+        if (new_fd == -1){
+            syslog(LOG_ERR , "accept\n");
+            // Try waiting for a connect() request from another remote
+            continue;
+        } 
+
+        // Print IP address
+        // The inet_toa function requires a sockaddr_in structure
+        // Reference: https://stackoverflow.com/questions/3060950/how-to-get-ip-address-from-sock-structure-in-c
+        struct sockaddr_in * ip_client = (struct sockaddr_in *)&client_addr;
+        syslog(LOG_INFO , "Accepted connection from %s\n", inet_ntoa(ip_client->sin_addr));  
+
+        while(!newline_flag){
+            byte_count = recv(new_fd , recv_buffer , BUFFER_SIZE , 0);
+            // printf("Byte count:%d\n", byte_count);
+            if ( byte_count == -1){
+                syslog(LOG_ERR , "recv\n");
+                return -1;
+            }
+            else{
+                final_size++;
+                for(datapacket = 0; datapacket < BUFFER_SIZE; datapacket++){
+                    if(recv_buffer[datapacket] == '\n'){
```

Comment:
@KhyatiSatta what would happen if `recv` received multiple packets or a part of another packet?  Are you guaranteed `recv` always will give you data on packet boundaries?